client/state/stats/lists/utils.js

```diff
@@ -300,6 +300,26 @@ export function parseUnitPeriods( unit, period ) {
 	}
 }
 
+export function parseStoreStatsReferrers( payload ) {
+	if ( ! payload || ! payload.data || ! Array.isArray( payload.data ) ) {
```

Comment:
I suppose we could also add a defensive check for `! payload.fields` too. Though guessing if that is missing, data is probably missing too.